test/functional/system/rounds.js

```diff
@@ -0,0 +1,1691 @@
+const constants = require('../../../helpers/constants');
+const slots = require('../../../helpers/slots');
+const Bignum = require('../../../helpers/bignum');
+const accountsFixtures = require('../../fixtures/accounts');
+const roundsFixtures = require('../../fixtures/rounds').rounds;
+const randomUtil = require('../../common/utils/random');
+
+describe('rounds', () => {
+	let library;
+	let keypairs;
+	let validateAccountsBalancesQuery;
+	let generateDelegateListPromise;
+
+	const Queries = {
+		validateAccountsBalances: () => {
+			return library.db.query(validateAccountsBalancesQuery);
+		},
+		getAccounts: () => {
+			return library.db.query('SELECT * FROM mem_accounts');
+		},
+		getDelegates: () => {
+			return library.db.query(
+				'SELECT * FROM mem_accounts m LEFT JOIN delegates d ON d.username = m.username WHERE d."transactionId" IS NOT NULL'
+			);
+		},
+		getDelegatesForList: () => {
+			return library.db.query(
+				`SELECT "publicKey", vote FROM mem_accounts ORDER BY vote DESC, "publicKey" ASC LIMIT ${
+					slots.delegates
+				}`
+			);
+		},
+		getFullBlock: height => {
+			return library.db
+				.query('SELECT * FROM full_blocks_list WHERE b_height = ${height}', {
+					height,
+				})
+				.then(rows => {
+					// Normalize blocks
+					return library.modules.blocks.utils.readDbRows(rows);
+				});
+		},
+		getBlocks: round => {
+			return library.db.query(
+				'SELECT * FROM blocks WHERE CEIL(height / 101::float)::int = ${round} ORDER BY height ASC',
+				{ round }
+			);
+		},
+		getRoundRewards: round => {
+			return library.db
+				.query(
+					'SELECT ENCODE("publicKey", \'hex\') AS "publicKey", SUM(fees) AS fees, SUM(reward) AS rewards FROM rounds_rewards WHERE round = ${round} GROUP BY "publicKey"',
+					{ round }
+				)
+				.then(rows => {
+					const rewards = {};
+					_.each(rows, row => {
+						rewards[row.publicKey] = {
+							publicKey: row.publicKey,
+							fees: row.fees,
+							rewards: row.rewards,
+						};
+					});
+					return rewards;
+				});
+		},
+		getVoters: () => {
+			return library.db.query(
+				'SELECT "dependentId", ARRAY_AGG("accountId") FROM mem_accounts2delegates GROUP BY "dependentId"'
+			);
+		},
+	};
+
+	before(done => {
+		validateAccountsBalancesQuery = new QueryFile(
+			path.join(
+				__dirname,
+				'../common/sql/rounds/validate_accounts_balances.sql'
+			),
+			{ minify: true }
+		);
+
+		// Set rewards start at 150-th block
+		constants.rewards.offset = 150;
+		application.init(
+			{ sandbox: { name: 'lisk_functional_rounds' } },
+			(err, scope) => {
+				library = scope;
+
+				generateDelegateListPromise = Promise.promisify(
+					library.modules.delegates.generateDelegateList
+				);
+
+				done(err);
+			}
+		);
+	});
+
+	after(done => {
+		application.cleanup(done);
+	});
+
+	function getMemAccounts() {
+		return Queries.getAccounts().then(rows => {
+			return _.cloneDeep(normalizeMemAccounts(rows));
+		});
+	}
```

Comment:
Yes, as expectations in those tests should be independent from application logic - for example in case there will be bug in `library.db.accounts.list()`.